src/python/impactx/dashboard/Analyze/plotsMain.py

```diff
@@ -0,0 +1,249 @@
+"""
+This file is part of ImpactX
+
+Copyright 2024 ImpactX contributors
+Authors: Parthib Roy, Axel Huebl
+License: BSD-3-Clause-LBNL
+"""
+
+import asyncio
+import contextlib
+import io
+import os
+
+from trame.widgets import matplotlib, plotly, vuetify
+
+from ..trame_setup import setup_server
+from .analyzeFunctions import AnalyzeFunctions
+from .plot_ParameterEvolutionOverS.overS import line_plot_1d
+from .plot_PhaseSpaceProjections.phaseSpace import run_simulation
+
+server, state, ctrl = setup_server()
+
+# -----------------------------------------------------------------------------
+# Plotting
+# -----------------------------------------------------------------------------
+
+
+# Call plot_over_s
+def plot_over_s():
+    """
+    Generates a plot.
+    """
+
+    fig = line_plot_1d(state.selected_headers, state.filtered_data)
+    ctrl.plotly_figure_update(fig)
+
+
+PLOTS = {
+    "Plot Over S": plot_over_s,
+    "Phase Space Plots": None,
+}
+
+# -----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
+
+
+def available_plot_options(simulationClicked):
+    """
+    Displays plot_options for users based on status of simulation.
+    :param simulationClicked (bool): status of simulation status
+    :return: list of plot_options for users
+    """
+
+    if simulationClicked:
+        return list(PLOTS.keys())
+    else:
+        return ["Run Simulation To See Options"]
+
+
+def load_dataTable_data():
+    """
+    Loads and processes data from combined beam and reference particle files.
+    """
+
+    if not os.path.exists(REDUCED_BEAM_DATA) or not os.path.exists(REF_PARTICLE_DATA):
+        ctrl.terminal_print(
+            "Diagnostics files are missing. Please ensure they are in the correct directory."
+        )
+        return
+
+    combined_files = AnalyzeFunctions.combine_files(
+        REDUCED_BEAM_DATA, REF_PARTICLE_DATA
+    )
+    combined_files_data_converted_to_dictionary_format = (
+        AnalyzeFunctions.convert_to_dict(combined_files)
+    )
+    data, headers = combined_files_data_converted_to_dictionary_format
+    state.all_data = data
+    state.all_headers = headers
+
+
+# -----------------------------------------------------------------------------
+# Defaults
+# -----------------------------------------------------------------------------
+
+CURRENT_DIR = os.getcwd()
+DIAGS_DIR = os.path.join(CURRENT_DIR, "diags")
+
+REDUCED_BEAM_DATA = os.path.join(DIAGS_DIR, "reduced_beam_characteristics.0.0")
+REF_PARTICLE_DATA = os.path.join(DIAGS_DIR, "ref_particle.0.0")
```

Comment:
Use this to read those files:
```py
def read_file(file_pattern):
    for filename in glob.glob(file_pattern):
        df = pd.read_csv(filename, delimiter=r"\s+")
        if "step" not in df.columns:
            step = int(re.findall(r"[0-9]+", filename)[0])
            df["step"] = step
        yield df

def read_time_series(file_pattern):
    """Read in all CSV files from each MPI rank (and potentially OpenMP
    thread). Concatenate into one Pandas dataframe.

    Returns
    -------
    pandas.DataFrame
    """
    return pd.concat(
        read_file(file_pattern),
        axis=0,
        ignore_index=True,
    )  # .set_index('id')
```
and then
```suggestion
REDUCED_BEAM_DATA = read_time_series(
    DIAGS_DIR + "/reduced_beam_characteristics.*"
)
REF_PARTICLE_DATA = read_time_series(DIAGS_DIR + "/ref_particle.*")
```

Reply:
I'm not too sure what the difference of having these new functions would do. Could you please explain? I have some other functions in analyzeFunctions that merge the dataframes into one

Reply:
Ah I see. Great, that works.

Small update: note that these files end in .number or .number.othernumber depending on running with or without MPI. Write the globbing general enough to be able to read either naming scheme please